### DISCUSSION TITLE
Find a project to contribute to

### DISCUSSION THREAD

**Discussion by d-wrede** [community/community#40390] [New to GitHub]
Hello everybody,
nice to meet you here. I'm new to github community, though have done scientific coding for a few years in Python and other languages, also applying git-structure. I would like to get into software development and believe that coding is best learned by coding! (more or less) So now I am searching for a project to contribute to, connected to a community, I might learn from. I read [this article](https://docs.github.com/en/get-started/exploring-projects-on-github/finding-ways-to-contribute-to-open-source-on-github) on github to find a project to contribute to, which already gives good hints, where to start - the projects are sorted by topics and I can see if they do have any "first good issues" to solve. 
It is easy to match the language "Python", which might be my starting point. But I'm still overwhelmed by all the projects and opportunities and am wondering how to match other criterions like:

1. Beginner-friendly project in size and complexity - so that I'm actually able to understand it
2. Beginner-friendly project community - and people who actually know how to code and like to share it
3. object oriented code
4. (renewable) energy related (nice to have), since I'm coming from that field

My current skillset is Python (NumPy, Matplotlib, MPI4Py), Delphi/TurboPascal and a little C(++). I would be glad to get more into software development in general and object oriented programming, but am also interested in pandas, SQL and (sooner or later) Javascript. Do you have any advice for how to find the right challenge/project? 
Thanks in advance! ;-)
LevinOX

**Comment by jfullstackdev** [comment]
this is where I started, 

https://github.com/pH-7/Simple-Java-Calculator

though it's Java, well, it's still worth learning, it's one of the most sought job in coding

> that coding is best learned by coding

yup, that's true, as Dennis Ritchie said, the creator of C language

![image](https://user-images.githubusercontent.com/47092464/204707772-eb84eb16-6858-438c-b003-6341bcb461fe.png)

as for today, maybe you want to learn full stack, in the Philippines, since there is no great budgeting most solutions companies will look for a full stack, whether it is PHP or Java, so PHP Laravel or Java Spring Boot, it's not enough to simply learn coding though it's the hardcore
